app/controllers/api/undo_controller.rb

```diff
@@ -0,0 +1,78 @@
+module Api
+  class UndoController < ApiController
+    def undo
+      case params[:undoable_action]
+      when 'change_next_page'
+        session[:undo] = {
+          action: 'undo',
+          undoable_action: 'change_next_page',
+          text: t('actions.undo_redo.undo_change_next_page')
+        }
+      when 'move'
+        session[:undo] = {
+          action: 'undo',
+          undoable_action: 'move',
+          text: t('actions.undo_redo.undo_move')
+        }
+      end
+      previous
```

Comment:
can we name this something a little more descriptive please

Reply:
Changed to call_previous_version